app/components/publish-actions.tsx

```diff
@@ -56,16 +59,49 @@ const TriggeredPopover = (props: TriggeredPopoverProps) => {
   );
 };
 
-const Embed = ({ configKey, locale }: PublishActionProps) => {
+const Embed = ({ configKey, locale, chartRef }: PublishActionProps) => {
   const [embedIframeUrl, setEmbedIframeUrl] = useState("");
   const [embedAEMUrl, setEmbedAEMUrl] = useState("");
+  const [responsiveIframe, setResponsiveIframe] = useState(true);
+  const [iframeHeight, setIframeHeight] = useState(640);
+
+  useEffect(() => {
+    if (chartRef?.current) {
+      const height = chartRef?.current?.getBoundingClientRect().height || 640;
```

Comment:
Super-nit: You can remove optional chaining operator from `chartRef.current.getBoundingClientRect()` 👀

Reply:
Another nit: I'd also round up the height, otherwise we have height="281.734375px" inside the iframe, not a huge deal, but could be nicer to have a whole number 😅